spec/workers/medium_removal_worker_spec.rb

```diff
@@ -1,15 +1,50 @@
+# frozen_string_literal: true
+
 require 'spec_helper'
 
 RSpec.describe MediumRemovalWorker do
-  it 'should delete the given src path' do
-    expect(MediaStorage).to receive(:delete_file)
-    subject.perform('test/path.txt')
+  subject(:worker) { described_class.new }
+
+  let(:medium_src) { 'test/path.txt' }
+
+  it 'deletes the given src path' do
+    allow(MediaStorage).to receive(:delete_file)
+    worker.perform(medium_src)
+    expect(MediaStorage).to have_received(:delete_file).with(medium_src, {})
   end
 
-  it 'should skip any access denied media paths' do
+  it 'ignores any missing s3 (access denied) media paths' do
     allow(MediaStorage)
       .to receive(:delete_file)
-      .and_raise(Aws::S3::Errors::AccessDenied.new(:s3, "fake denied"))
-    expect { subject.perform('test/path.txt') }.not_to raise_error
+      .and_raise(Aws::S3::Errors::AccessDenied.new(:s3, 'fake denied'))
+    expect { worker.perform(medium_src) }.not_to raise_error
+  end
+
+  it 'ignores any missing azure (BlobNotFound (404)) media paths' do
+    require 'azure/core/http/http_error'
+    response = instance_double('Azure::Core::Http::HTTPResponse', uri: 'fake-uri', status_code: 500, body: '', reason_phrase: '')
+    allow(MediaStorage).to receive(:delete_file).and_raise(Azure::Core::Http::HTTPError, response)
+    expect { worker.perform(medium_src) }.not_to raise_error
+  end
+
+  it 'does not modify the storage path for the object store' do
```

Comment:
Would be helpful to specify that storage path is not modified specifically for the case where we are not using the azure adapter. Otherwise might be confusing for us when we're looking back on this to read this as default behavior.
```suggestion
  it 'does not modify the storage path for the object store when not using azure adapter' do
```